### ISSUE TITLE
ITKSetPython3Vars.cmake ignores virtualenvs and other issues

### ISSUE THREAD

**Issue by brad-t-moore** [InsightSoftwareConsortium/ITK#2068]
### Description

[ITKSetPython3Vars ](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CMake/ITKSetPython3Vars.cmake#L38-L53) ignores virtualenvs with CMake > 3.15.0.  It also has suspicious use of CMAKE FindPython internal variables.  I think we should define the expected user interaction with ITK and specifying the Python version, as well as fix some bugs.  @thewtex @hjmjohnson 

### Steps to Reproduce

We currently have special cases for CMake <= 3.12.  There was a significant change to FindPython in CMake 3.15.  Due to our CMake Minimum version, [CMP0094 ](https://cmake.org/cmake/help/git-stage/policy/CMP0094.html) (which allows virtualenvs to have precedence) is turned undefined (OFF).

Activate a virtualenv that is not the highest version of Python found on the system.  Try and build ITK.

### Expected behavior

For CMake < 3.12, keep current logic.

For CMake 3.12 - 3.15 ??? (the API for [FindPython 3.14](https://cmake.org/cmake/help/v3.14/module/FindPython.html) is different and doesn't mention virtualenvs)

For CMake > 3.15, enable [CMP0094 ](https://cmake.org/cmake/help/git-stage/policy/CMP0094.html) and get rid of references to _specified_Python3_EXECUTABLE

Document how we expect users to define Python version.  Or expose the allowed hints through external cache variables.

### Actual behavior

For CMake > 3.15, with CMP0094 undefined, [FindPython 3.19](https://cmake.org/cmake/help/v3.19/module/FindPython.html) will only return the latest version of Python found on the system (whether it has the Development/debug build of Python or not).

```
else()
  if(PYTHON_DEVELOPMENT_REQUIRED)
    if(DEFINED Python3_EXECUTABLE)
      set(_specified_Python3_EXECUTABLE ${Python3_EXECUTABLE}) # <--- where is _specified_Python3_EXECUTABLE defined
    endif()
    find_package(Python3 COMPONENTS Interpreter Development)
    set(Python3_EXECUTABLE ${_specified_Python3_EXECUTABLE} CACHE INTERNAL
      "Path to the Python interpreter" FORCE) # <--- What are we doing here?  This is by default blank, so we force an overwrite of Python3_EXECUTABLE
  else()
    find_package(Python3 COMPONENTS Interpreter)
  endif()
  if(NOT Python3_EXECUTABLE AND _Python3_EXECUTABLE) # <--- This always executes because of the overwrite above
    set(Python3_EXECUTABLE ${_Python3_EXECUTABLE} CACHE INTERNAL
      "Path to the Python interpreter" FORCE) # <--- We are now referring to a private CMAKE variable?
  endif()
endif()
```

```none
```

**Comment by thewtex** [comment]
The `FindPython` CMake module is broken in many ways. We should not spend effort trying to support older versions of it, because it has numerous other issues. If it helps to enable `CMP0094` for the current version of CMake, then let's do that.

**Comment by brad-t-moore** [comment]
@thewtex OK, so for the oddball CMake 3.12 - 3.15.  Should I push the behavior back to using FindPythonInterp (i.e. increase the first if-statement from <= 3.12 to <= 3.15, or should I maintain the current behavior (using CMake FindPython but it won't have the option to set CMP0094).

Is there any user documentation that should be updated for this patch?

**Comment by thewtex** [comment]
> OK, so for the oddball CMake 3.12 - 3.15. Should I push the behavior back to using FindPythonInterp (i.e. increase the first if-statement from <= 3.12 to <= 3.15,

@brad-t-moore yes, please

> Is there any user documentation that should be updated for this patch?

I do not think so.

Thank you so much for your work on this.

**Comment by hjmjohnson** [comment]
@thewtex For python-support should we just require cmake >= 3.16?  I know it is a very new version of cmake, but that could greatly minimize our maintenance efforts.

**Comment by brad-t-moore** [comment]
Closing this as we got the Windows CI working but created an issue for updating the version check #2082

Fixed by PR #2072
